### PR TITLE
dirty-flag: avoid always reconfigure

### PR DESCRIPTION
**Pull request by daschuer**
This is an attempt to avoid reconfigure. 
The pending issue is that it is reconfigured twice ins some cases.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1676,6 +1672,32 @@ get_target_property(BUILD_FLAGS mixxx-lib COMPILE_OPTIONS)
...
 
+configure_file(cmake/gitdescribecache.in gitdescribecache @ONLY)
+
+add_custom_target(
+    update_git_describe ALL
+    DEPENDS git_describe_ceck ${CMAKE_BINARY_DIR}/CPackConfig.cmake
+)
+
+add_custom_command(
+    OUTPUT git_describe_ceck # never produced, always run
```
Comment (Holzhaus) [inline CMakeLists.txt:1683]: typo

**Review by Holzhaus — COMMENTED** [review]
Thanks, but I find this hard to read.

I have another idea that even allows using git describe (including the dirty flag) in the package file name and doesn't reconfigure at all (unless CMakeLists.txt is edited). Let me check if that works one last time :D Hopefully we can finally get rid of this annoying problem.

**Comment by daschuer** [comment]
> I have another idea that even allows using git describe (including the dirty flag) in the package file name

That works here. 

> and doesn't reconfigure at all (unless CMakeLists.txt is edited). 

That would be nice. 

> Let me check if that works one last time :D Hopefully we can finally get rid of this annoying problem.

Can you give some details? 

This does not jet work with ninja, because ninja does the file change check only once.

**Comment by Holzhaus** [comment]
> > Let me check if that works one last time :D Hopefully we can finally get rid of this annoying problem.
> 
> Can you give some details?

What I meant was one last attempt to fix this git-info problem in CMakeLists in general (considering we now have 5 open PRs concerning that).

Please have a look at #3841.

**Pull request closed without merging**